### PR TITLE
maint: Require automake >= 1.16.5 in release.pl

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

This will ensure tarballs are generated with up-to-date config.guess
files for better cross compilation support. Fixes https://github.com/pmodels/mpich/issues/4987.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
FYI, `automake-1.16.5` was released on 2021-10-03.

**Review by hzhou — APPROVED** [review]
LGTM

File: maint/release.pl [hunk 1]
```diff
@@ -160,9 +160,11 @@ sub run_cmd
...
 
 ## NOTE: Different autotools versions may result in accidental ABI chanages.
-## For flexibility, we no longer enforce the check for specific versions.
-## Always double check using the ABI compatibility tool before the final release.
-
+## For flexibility, we no longer enforce the check for specific versions of
+## autoconf and libtool. We do require a minimum version of automake so tarballs
+## contain up-to-date config.guess files. Always double check using the ABI
+## compatibility tool before the final release.
+check_autotools_version("automake", "1.16.5")
```
Comment (hzhou) [inline maint/release.pl:167]: Oops, missing semicolon @raffenet

Comment (raffenet) [reply maint/release.pl:167]: Darn. Will fix.

**Pull request merged**